scripts/bundle-rhmi-operators.sh

```diff
@@ -134,33 +136,37 @@ generate_index() {
 push_index() {
   VERSIONS_TO_PUSH=$1
   bundles=""
-  for VERSION in $(echo "$VERSIONS_TO_PUSH" | sed "s/,/ /g")
-  do
-      bundles=$bundles"$REG/$ORG/${OLM_TYPE}-bundle:$VERSION,"
+  for VERSION in $(echo "$VERSIONS_TO_PUSH" | sed "s/,/ /g"); do
+    bundles=$bundles"$REG/$ORG/${OLM_TYPE}-bundle:$VERSION,"
   done
   # remove last comma
   bundles=${bundles%?}
 
-  NEWEST_VERSION="$( cut -d ',' -f 1 <<< "$VERSIONS_TO_PUSH" )"
+  NEWEST_VERSION="$(cut -d ',' -f 1 <<<"$VERSIONS_TO_PUSH")"
   INDEX_IMAGE=$REG/$ORG/${OLM_TYPE}-index:$NEWEST_VERSION
 
   # Checks if script should preserve existing index images at $REG/$ORG/${OLM_TYPE}-index:$NEWEST_VERSION
   # If the index image doesn't already exist, it will be created
   # If the index image does already exist, the script continues
-  if [ "$OVERWRITE" = false ] ; then
-    exists="$($BUILD_TOOL manifest inspect "$INDEX_IMAGE" > /dev/null ; echo $?)"
+  if [ "$OVERWRITE" = false ]; then
+    exists="$(
+      $BUILD_TOOL manifest inspect "$INDEX_IMAGE" >/dev/null
+      echo $?
+    )"
     # exists will by 0 on success (image does exist) and 1 on failure (image does not exist)
-    if [ "$exists" -eq "0" ] ; then
+    if [ "$exists" -eq "0" ]; then
       printf "Not building or pushing this index image because it already exists and OVERWRITE is set to false: %s\n" "$INDEX_IMAGE"
       INDEX="$INDEX $INDEX_IMAGE"
       return
     fi
   fi
 
   opm index add \
-      --bundles "$bundles" \
-      --container-tool "$BUILD_TOOL" \
-      --tag "$REG/$ORG/${OLM_TYPE}-index:$NEWEST_VERSION"
+    --bundles "$bundles" \
+    --container-tool "$BUILD_TOOL" \
+    --generate
```

Comment:
I am interested to what the `--generate` does. I guess it is creating the index.Dockerfile mentioned on L169. This raises two questions.
1. Are you sure the creation of the file while always be in the same location?
2. When running this script locally is it going to be up the user to clean up this file after completion?

Reply:
1. Yeah, I believe it would always be in the same location. The docs mentions it is generated in the directory where `opm` is ran. From what I have seen / checked, it is created in the temp bundle directory 🤔 
  * https://github.com/operator-framework/operator-registry/blob/master/docs/design/opm-tooling.md#add-1 
  * https://github.com/integr8ly/integreatly-operator/blob/master/scripts/bundle-rhmi-operators.sh#L65-L69
2. No, it would be cleaned up by the script. So as above it is created in the work area temp directory and this directory is cleaned up at the end of the script
  * https://github.com/integr8ly/integreatly-operator/blob/master/scripts/bundle-rhmi-operators.sh#L180-L183